pulp_python/app/utils.py

```diff
@@ -414,6 +418,46 @@ def write_simple_detail(project_name, project_packages, streamed=False):
     return detail.stream(**context) if streamed else detail.render(**context)
 
 
+def write_simple_index_json(project_names):
+    """Writes the simple index in JSON format."""
+    projects = [{"_last-serial": PYPI_SERIAL_CONSTANT, "name": name} for name in project_names]
+
+    return {
+        "meta": {"api-version": PYPI_API_VERSION, "_last-serial": PYPI_SERIAL_CONSTANT},
+        "projects": projects,
+    }
+
+
+# todo: fields
+def write_simple_detail_json(project_name, project_packages):
+    """Writes the simple detail page in JSON format."""
+    files = []
+    for filename, url, sha256 in project_packages:
```

Comment:
It might make sense to change project_packages to a list of dictionaries, or namedtuples, and then modify the simple_detail template accordingly.